Mathlib/RingTheory/DedekindDomain/Ideal.lean

```diff
@@ -255,7 +255,10 @@ theorem isDedekindDomainInv_iff [Algebra A K] [IsFractionRing A K] :
 theorem FractionalIdeal.adjoinIntegral_eq_one_of_isUnit [Algebra A K] [IsFractionRing A K] (x : K)
     (hx : IsIntegral A x) (hI : IsUnit (adjoinIntegral A⁰ x hx)) : adjoinIntegral A⁰ x hx = 1 := by
   set I := adjoinIntegral A⁰ x hx
-  have mul_self : I * I = I := by apply coeToSubmodule_injective; simp [I]
+  have mul_self : IsIdempotentElem I := by
+    apply coeToSubmodule_injective;
+    simp [I]
+    rw [Subalgebra.mul_self (Algebra.adjoin A {x})]
```

Comment:
You shouldn't have `rw` after unsqueezed `simp`

Reply:
Done, thanks.